internal/resource/list.go

```diff
@@ -254,32 +260,34 @@ func (l *list) load(informer *wa.Informer, ns string) (Columnars, error) {
 }
 
 func (l *list) fetchResource(informer *wa.Informer, r interface{}, ns string) (Columnar, error) {
-	var err error
-
 	res := l.resource.New(r)
+
 	switch o := r.(type) {
 	case *v1.Node:
 		fqn := MetaFQN(o.ObjectMeta)
-		nmx, err := informer.Get(wa.NodeMXIndex, fqn, metav1.GetOptions{})
-		if err == nil {
+		if nmx, err := informer.Get(wa.NodeMXIndex, fqn, metav1.GetOptions{}); err != nil {
+			return res, err
+		} else {
 			res.SetNodeMetrics(nmx.(*mv1beta1.NodeMetrics))
 		}
```

Comment:
I actually prefer the pattern
```go
nmx, err := informer.Get()
if err != nil {
    return res, err
}
res.SetNodeMetrics(nmx.(*mv1beta1.NodeMetrics))
```

Declaring everything inside the  `if` creates a big single line which may be hard to read.

Reply:
Good point! Tho it's more about scoping than readability IMHO and it this case the scope is indeed irrelevant as we're in a switch.